src/Commands.cpp

```diff
@@ -394,3 +402,146 @@ void	handleModeCommand(Server& server, Client* client, const std::string& params
 	std::string modeMessage = ":" + client->getNickname() + " MODE " + channelName + " " + modeString + " " + modeParam + "\r\n";
 	channel->broadcast(modeMessage, client);
 }
+
+void handleKickCommand(Server& server, Client* kicker, const std::string& params) {
+	std::istringstream iss(params);
+	std::string channelName, targetNickname, reason;
+	iss >> channelName >> targetNickname;
+	
+	std::getline(iss, reason);
+	reason = reason.empty() ? "No reason specified" : reason.substr(1); // Remove leading space
+
+	if (channelName.empty() || targetNickname.empty()) {
+		server.sendError(kicker->getFd(), "461", "KICK :Not enough parameters");
+		return;
+	}
+
+	Channel* channel = server.getChannel(channelName);
+	if (!channel) {
+		server.sendError(kicker->getFd(), "403", channelName + " :No such channel");
+		return;
+	}
+
+	if (!channel->isMember(kicker)) {
+		server.sendError(kicker->getFd(), "442", channelName + " :You're not on that channel");
+		return;
+	}
+
+	if (!channel->isOperator(kicker)) {
+		server.sendError(kicker->getFd(), "482", channelName + " :You're not a channel operator");
+		return;
+	}
+
+	Client* targetClient = server.getClientByNickname(targetNickname);
+	if (!targetClient) {
+		server.sendError(kicker->getFd(), "401", targetNickname + " :No such nick");
+		return;
+	}
+
+	if (!channel->isMember(targetClient)) {
+		server.sendError(kicker->getFd(), "441", targetNickname + " " + channelName + " :Hi isn't on that channel");
+		return;
+	}
+
+	if (targetClient == kicker) {
+		server.sendError(kicker->getFd(), "485", " :You can't kick yourself");
+		return;
+	}
+
+	// Remove the client from the channel
+	channel->removeMember(targetClient);
+	// Notify all channel members about the kick
+	std::string kickMessage = ":" + kicker->getNickname() + " KICK " + channelName + " " + targetNickname + " :" + reason + "\r\n";
+	channel->broadcast(kickMessage);
+	// Notify the target client
+	server.sendMessage(targetClient->getFd(), kickMessage);
+}
```

Comment:
I've been using [this RFC protocol document](https://datatracker.ietf.org/doc/html/rfc1459) you sent me, because it seems like a really good basis for our project. While looking through it I found this:
![image](https://github.com/user-attachments/assets/b3e2bcbf-098b-46b8-865b-3148d745343a)
Should we handle it? @dlbltv

Reply:
@mortinso I don't think we should handle this case for KICK command. Since we don't have the ability to change the nickname. And it's not required in the subject as far as I've seen.

Reply:
Ok! You're right our version of `NICK` doesn't change the nickname.